pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -114,23 +111,35 @@ let
 
       _nugetDeps =
         if (nugetDeps != null) then
-          if lib.isDerivation nugetDeps then
+          if lib.isList nugetDeps then
             nugetDeps
           else
-            mkNugetDeps {
-              inherit name;
-              sourceFile = nugetDeps;
-            }
+            assert (lib.isPath nugetDeps);
+            callPackage nugetDeps { fetchNuGet = fetchNupkg; }
         else
-          throw "Defining the `nugetDeps` attribute is required, as to lock the NuGet dependencies. This file can be generated by running the `passthru.fetch-deps` script.";
+          [ ];
 
-      nugetDepsFile = _nugetDeps.sourceFile;
+      nugetDepsFile = if lib.isPath nugetDeps then nugetDeps else null;
 
       inherit (dotnetCorePackages) systemToDotnetRid;
     in
     # Not all args need to be passed through to mkDerivation
     # TODO: We should probably filter out even more attrs
-    removeAttrs args [ "nugetDeps" ]
+    removeAttrs args [
+      "nugetDeps"
+      "installPath"
+      "executables"
+      "projectFile"
+      "projectReferences"
+      "runtimeDeps"
+      "runtimeId"
+      "disabledTests"
+      "testProjectFile"
+      "buildType"
+      "selfContainedBuild"
+      "useDotnet"
+      "useAppHost"
+    ]
```

Comment:
:heart:

Reply:
This is a breaking change unfortunately.  `roslyn-ls` for example, was using `$projectFile`, and needed to be changed to `$dotnetProjectFiles`.

It's a little unintuitive that something like:

```
projectFile = "foo";
buildPhase = "echo $projectFile";
```

doesn't work.  @tie you did #339335.  What do you think?

Reply:
Can I ask, what's the motivation for prefixing dotnet-specific derivation args in the first place? 

I struggle to see how name conflicts would happen, for instance.

Reply:
For `buildDotnetModules` I agree.  I probably wouldn't have prefixed them.

For hooks like `dotnet-sdk-setup-hook` I'm trying to be careful about collisions, because I want that stuff to work well in a mixed build environment.  Ideally I'd like it to be as easy to build a dotnet project with `stdenv[NoCC]` as with `buildDotnetModule`.

Reply:
> For hooks like `dotnet-sdk-setup-hook` I'm trying to be careful about collisions, because I want that stuff to work well in a mixed build environment.

Makes sense. :+1: 

From my perspective, we have the verbose/prefixed names for anything handled by the dotnet hook, since that hook may be used with arbitrary stdenv derivations and therefore should avoid potential name collisions.

The arguments used by `buildDotnetModule` could therefore almost be thought of as aliases, since the verbosity is redundant in that context.

Looking at it from this perspective, I don't think it is such a bad thing for `buildDotnetModule` to pass all its arguments to stdenv.

On reflection I think this is the least surprising behaviour, even if it does feel a little messy _internally_; an end user (packager) would likely expect _any_ argument they've passed to a builder function to be also made available in stdenv.

> Ideally I'd like it to be as easy to build a dotnet project with `stdenv[NoCC]` as with `buildDotnetModule`.

That sounds great!

Reply:
> On reflection I think this is the least surprising behaviour, even if it does feel a little messy internally; an end user (packager) would likely expect any argument they've passed to a builder function to be also made available in stdenv.

One problem with that is sometimes, like with nugetDeps, there are negative consequences to including it.

Stripping them definitely feels wrong though.  Like either it should be two separate attrset arguments, or one as an attribute in the other.